packages/components/src/ActionList/ActionList.tsx

```diff
@@ -170,8 +190,8 @@ export const ActionList = styled(ActionListLayout)<ActionListProps>`
   }
 
   ${ActionListHeader} {
-    padding-left: ${({ canSelect }) =>
-      canSelect ? actionListCheckboxWidth : undefined};
+    padding-left: ${({ canSelect, canSelectAll }) =>
+      canSelect && !canSelectAll ? actionListCheckboxWidth : undefined};
```

Comment:
I think a guard should be put in place to prevent this combination check – for `canSelectAll` to be `false` if `canSelect=false` so the logic doesn't need to be held in more than one place.

```suggestion
    padding-left: ${({ canSelect, canSelectAll }) =>
      !canSelectAll && actionListCheckboxWidth};
```

Reply:
That makes sense! I ended up changing the lines where the context object is created. The context's `canSelectAll` property is now reliant on both the `canSelect` and `canSelectAll` props, so any other component using the context will not have to performa an additional check for `canSelect`. See 6379e2d.

However, for this chunk of styling in particular, I _think_ we do need the combination check  (it's meant to handle the case where you have `canSelect` true but `canSelectAll` false). We want to have the header have a chunk of padding-left only in the case where the row's have checkboxes, and the header doesn't.

Let me know if I'm misunderstanding something here though with the suggested commit or the suggestion in general.